### PR TITLE
OUT-654 | Improving task details opening experience

### PR DESCRIPTION
**Pull request by rrojan**
### Tasks

- [OUT-654-re | Improving task details opening experience](https://linear.app/copilotplatforms/issue/OUT-654/improving-task-details-opening-experience-video-shows-laggy-navigation)

### Changes

- [x] Add assignee data to `/api/task/:id` endpoint so that additional fetching and filtering is not required.
- [x] Removed all the unnecessary fetch calls from the `/details` route
- [x] Adds a top loader using `react-progressload` component
- [x] Refactors the `Realtime` approach 
- [x] Adds suspense boundary to individual fetch calls wherever feasible and adds skeleton loaders for lazy loading

### Testing criteria 
- Load the app and notice that initial load time has been reduced
- Notice that the assignee name in the `TaskCard` loads later with skeleton loaders displayed initially
- Navigate to the `/details` route and notice the load time has been reduced
- Notice that the `Sidebar` component loads lazily with the skeleton loader displayed initially

### CODE REVIEW

File: src/app/api/tasks/tasks.service.ts
Comment (sazanrjb) [inline src/app/api/tasks/tasks.service.ts:165]: return type missing?

**Review by rrojan — COMMENTED** [review]
Please fix merge conflicts @aatbip

**Review by saugatmaharjan — APPROVED** [review]

**Review by aatbip — APPROVED** [review]
@rrojan created the PR so he can't approve. Approving by myself after his internal approval.

**Pull request merged**
